src/realm/set.hpp

```diff
@@ -97,6 +97,48 @@ class Set final : public CollectionBaseImpl<SetBase> {
         }
     }
 
+    template <class Rhs>
+    bool is_subset_of(const Rhs&) const;
+
+    template <class It1, class It2>
```

Comment:
Why 2 template parameters? The 2 iterators will always be of the same type.

Reply:
Because the standard algorithms support iterators of different types. It's meant to support iterator pairs where the end is a sentinel.